feat: Adding basic main portrait layout

alexjba opened the pull request:
### What does the PR do

Introducing `LayoutChooser` component from qml tests: https://codereview.qt-project.org/c/qt/qtdeclarative/+/456325/54/tests/manual/quicklayouts/layoutItemProxy/LayoutChooser.qml

The LayoutChooser is used to change the "StatusSectionLayout" from a SplitView to a SwipeView in portrait mode whenever the width is less than the implicitWidth.

Other changes:
Maintain compoatibility with qt5
Basic navigastion on item selection (channels, contacts, settings, accounts)

<!-- Fill in the relevant information below to help us evaluate your proposed changes. -->

### Affected areas

App layout, navBar
<!-- List the affected areas (e.g wallet, browser, etc..) -->

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

https://github.com/user-attachments/assets/59f963b2-d87c-47e6-9637-416f1651ad8f

alexjba committed:
feat: Adding basic main portrait layout

Introducing `LayoutChooser` component from qml tests: https://codereview.qt-project.org/c/qt/qtdeclarative/+/456325/54/tests/manual/quicklayouts/layoutItemProxy/LayoutChooser.qml

The LayoutChooser is used to change the "StatusSectionLayout" from a SplitView to a SwipeView in portrait mode whenever the width is less than the implicitWidth.

Other changes:
Maintain compoatibility with qt5
Basic navigastion on item selection (channels, contacts, settings, accounts)

caybro reviewed (commented):
The new component is probably not working with Qt5, as the failing ui-tests suggest:
```
[2025-05-29T09:52:51.451Z] Running: StatusQ SanityChecker
[2025-05-29T09:52:51.817Z] (file:///home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-18012/ui/StatusQ/src/StatusQ/Layout/StatusSectionLayoutPortrait.qml:150:37: Expected token `numeric literal'
[2025-05-29T09:52:51.817Z]             property list<Item> items: []
[2025-05-29T09:52:51.817Z]                                         ^)
[2025-05-29T09:52:51.817Z] (file:///home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-18012/ui/StatusQ/src/StatusQ/Layout/LayoutChooser.qml:11:36: Unexpected token `]'
[2025-05-29T09:52:51.817Z]         property list<bool> criteria: []
[2025-05-29T09:52:51.818Z]                                        ^, file:///home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-18012/ui/StatusQ/src/StatusQ/Layout/LayoutChooser.qml:12:14: Syntax error
[2025-05-29T09:52:51.818Z]         property list<var> layoutChoices: []
[2025-05-29T09:52:51.818Z]                  ^)
[2025-05-29T09:52:51.818Z] (file:///home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-18012/ui/StatusQ/src/StatusQ/Layout/StatusSectionLayoutLandscape.qml: Library import requires a version)
```

caybro commented on ui/StatusQ/src/StatusQ/Components/StatusChatList.qml:
```diff
@@ -227,6 +228,7 @@ Item {
                             if (!statusChatListItem.selected) {
                                 root.chatItemSelected(statusChatListItem.categoryId, statusChatListItem.chatId)
                             }
+                            root.chatItemClicked(statusChatListItem.chatId)
```
How is this different from the other signal, `chatItemSelected`?

caybro commented on ui/StatusQ/src/StatusQ/Layout/LayoutChooser.qml:
Shouldn't this be a Qt6 component?

alexjba replied:
it is a Qt6 component. Just that I couldn't find a nice way of keeping it in the +qt6 folder. We need a qmldir for it to keep it in the +qt6, meaning that we'll end-up with a different qmldir for qt6. Tried to use an extended qmldir for +qt6, but it didn't work.

Same case with the `StatusSectionLayoutLandscape` and `StatusSectionLayoutPortrait`

caybro commented on ui/StatusQ/src/StatusQ/Layout/StatusSectionLayoutLandscape.qml:
```diff
@@ -0,0 +1,236 @@
+import QtQuick 2.13
+import QtQuick.Layouts
+import QtQuick.Controls 2.13
```
Some weird combo of imports here

caybro commented on ui/StatusQ/src/StatusQ/Layout/StatusSectionLayoutPortrait.qml:
```diff
@@ -0,0 +1,239 @@
+import QtQuick
+import QtQuick.Layouts
+import QtQuick.Controls
```
Versions?

caybro commented on ui/app/AppLayouts/Chat/views/ChatView.qml:
```diff
@@ -189,58 +189,55 @@ StatusSectionLayout {
         return root.chatContentModule.chatDetails.isUsersListAvailable
     }
 
-    rightPanel: Component {
```
Any prob with this not being a Component? :thinking:

alexjba replied:
I don't see any. What we have in the qt5 layout is a Control with a Loader for this component that's always active. The `showRightPanel` flag controls the Control opacity and visibility.

With qt6 we'll have an Item as input for the rightPanel. If the `showRightPanel` flag is false it will behave in a similar way, hiding it. The code is almost identical, just replacing an always active `Loader` with `LayoutItemProxy`

caybro reviewed (approved):
LGTM, very nice work for a first iteration!

alexjba committed:
fix: Fixing wallet footer

alexjba committed:
fix: Updating the breakpoints to use the app minimum width and height

The app goes into portrait mode the height is higher than the width and the width is smaller than the window minimum width

alexjba commented:
> > @caybro please have another look. Had to resolve quite a few conflicts with the Shell
> 
> Looks good, although I can only see one line touched in AppMain.qml that mentions shell?

Got me worried for a second! But yeah, you're right. Ignoring the whitespace it's only a line. 😄 

The main change is that the shell bar (along with the app nav bar) now lives in AppMain and is shared in between StatusSectionLayouts. It's no longer in the `StatusMainLayout`

micieslak reviewed (approved):
Great stuff! I played with it on desktop and works really nice. We could consider enabling it on desktop as well. Minimal width could be smaller and trigger re-layouting when shrunk.

caybro reviewed (approved):
Needs a rebase but otherwise LGTM :)

noeliaSD commented on ui/app/AppLayouts/Chat/views/ChatView.qml:
```diff
@@ -188,50 +188,47 @@ StatusSectionLayout {
-                chatDetails: !!root.chatContentModule ? root.chatContentModule.chatDetails : null
-                chatCommunitySectionModule: root.rootStore.chatCommunitySectionModule
-            }
+    rightPanel: UserListPanel {
```
This will confict with some recent changes I pushed I think!

alexjba replied:
that's fine. All I'm interested in is to drop the `Component` and use `UserListPanel` directly. I've just copied whatever is on master in `UserListPanel` body

noeliaSD reviewed (commented):
LGTM, just some minor questions / suggestions!

alexjba commented on ui/StatusQ/src/StatusQ/Layout/StatusSectionLayoutLandscape.qml:
```diff
@@ -0,0 +1,236 @@
+    }
+
+    Control {
+        SplitView.minimumWidth: (!!leftPanel) ? 304 : 0
```
yes, all this landscape section is meant to be just a duplicate section, based on what we had for qt5. I was intending to touch it as little as possible now. All your comments here are truly valid, but I'd do it in a following PR/task. WDYT?

alexjba committed:
chore: Add some comments to the StatusSectionLayoutPortrait explaining the wrapper caching
